tools/cli/internal/openapi/filter/extension.go

```diff
@@ -15,49 +15,102 @@ package filter
 
 import (
 	"log"
+	"time"
 
 	"github.com/getkin/kin-openapi/openapi3"
 	"github.com/mongodb/openapi/tools/cli/internal/apiversion"
 )
 
-type SunsetFilter struct {
+type ExtensionFilter struct {
 	oas      *openapi3.T
 	metadata *Metadata
 }
 
 const sunsetExtension = "x-sunset"
```

Comment:
Feel free to do it in a follow up PR but it may makes sense to add a comment explaining what the filter does since now performs two operations

Reply:
yep!